byway/src/main/webapp/recommendations.js

```diff
@@ -0,0 +1,194 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global google */
+/* global step:writable, interest:writable, region:writable, result:writable */
+
+if(document.readyState === 'loading') {
+  document.addEventListener('DOMContentLoaded', initialize);
+} else {
+  initialize();
+}
+
+let map;
+let placesService;
+
+// Measured in meters
+let RADIUS = 1000;
+let MIN_DISTANCE = 4000;
+
+let interests = ["park"];
+let destinations = new Set();
+
+// Contains google.maps.LatLng objects.
+// Used as a center point to search around a region.
+let regions = [];
+
+/**
+ * Initializes the webpage with a map and other google
+ * services. Creates a route between two endpoints.
+ */
+function initialize() {
+  // Modified version of Justine's implementation
+  var directionsService = new google.maps.DirectionsService();
```

Comment:
i think zach/landon had mentioned before to try not to use var, and use let or const instead. not sure if it is different when using the api but maybe try to let/const  to that for these variables

Reply:
Updated for all `var` types, also updated some `let` types